src/heitech.zer0mqXt.core/Infrastructure/SocketConfiguration.cs

```diff
@@ -16,47 +16,56 @@ internal abstract class SocketConfiguration : IEquatable<SocketConfiguration>
         ///<summary>
         /// Uses the BasicLogger implementation by default which only Logs to the console
         ///</summary>
-        public ILogger Logger { get; set; }
+        public ILogger Logger 
+        {
+            get
+            {
+                if (_logger == null)
+                    _logger = new BasicLogger();
 
+                return _logger;
```

Comment:
Could be a one-liner with `get => _logger ?? new BasicLogger();` - but it's just style.

Reply:
Now u r fucking with me right? That is exactly how it was before...

Reply:
Oh no!!! My bad ... aargh ... I meant to suggest the _null-coalescing assignment operator_:
`get => _logger ??= new BasicLogger();`